openstates/md/bills.py

```diff
@@ -165,10 +168,105 @@ def parse_bill_votes(self, doc, bill):
                 vote.add_source(href)
                 yield vote
 
+    def parse_bill_votes_new(self, doc, bill):
+        elems = doc.xpath('//a')
+
+        # MD has a habit of listing votes twice
+        seen_votes = set()
+
+        for elem in elems:
+            href = elem.get('href')
+            if (href and "votes" in href and href.endswith('pdf') and
+               ("Senate" in href or "House" in href) and href not in seen_votes):
+                seen_votes.add(href)
+                vote = self.parse_vote_pdf(href, bill)
+                vote.add_source(href)
+                yield vote
+
+    def parse_vote_pdf(self, vote_url, bill):
+        filename, response = self.urlretrieve(vote_url)
+        text = convert_pdf(filename, type='text').decode()
+        lines = text.splitlines()
+
+        if 'senate' in vote_url:
+            chamber = 'upper'
+        else:
+            chamber = 'lower'
+
+        date = dparse(lines[0].split('Date:')[1], fuzzy=True)
```

Comment:
We try to avoid `dateutil`, since when provided with messy legislative dates, it can make incorrect assumptions about how to parse them. Are the dates all in the same format, so that we could `datetime.datetime.strptime` them?

Reply:
In general, it's also ideal to not add any dependencies to the project. If this were to be absolutely necessary, the PR would need an explanation, and to touch `requirements.txt` as well.